From 6703b02f1bd6eb72f278dd93bcb47827f626ebe5 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 9 Jun 2016 14:13:39 +0000
Subject: [PATCH] Fix fexists function

build/Configured/src/rolegen.cpp: In function 'bool fexists(std::__cxx11::string)':
build/Configured/src/rolegen.cpp:54:10: error: cannot convert 'std::ifstream {aka std::basic_ifstream<char>}' to 'bool' in return
   return ifile;
          ^~~~~
---
 src/rolegen.cpp.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rolegen.cpp.in b/src/rolegen.cpp.in
index eb90368..3236af6 100755
--- a/src/rolegen.cpp.in
+++ b/src/rolegen.cpp.in
@@ -51,7 +51,7 @@ static bool isPrivilegedAppId(const char *appId)
 static bool fexists(string filename)
 {
   ifstream ifile(filename.c_str());
-  return ifile;
+  return ifile.fail();
 }
 
 static void checkDirStructure()
-- 
1.9.1

